.github/workflows/publish-container.yaml

```diff
@@ -16,41 +16,62 @@ jobs:
     permissions:
       contents: read
       packages: write
+    env:
+      tags: >-
+        ${{ github.ref == format('refs/heads/{0}', 'main') && 'latest' || '' }}
+        ${{ github.event_name == 'pull_request' && format('pr-{0}', github.event.number) || '' }}
+        ${{ github.event_name == 'push' && github.ref_name || '' }}
+        sha-${{ github.event_name == 'pull_request' && github.event.pull_request.head.sha || github.sha }}
     steps:
       - name: Checkout
         uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
         with:
           ref: ${{ github.event.pull_request.head.sha }}
-      - name: Docker meta
-        id: meta
-        uses: docker/metadata-action@8e5442c4ef9f78752691e2d8f8d19755c6f78e81 # v5.5.1
-        with:
-          images: |
-            ghcr.io/tkhq/ecr-proxy
-          tags: |
-            type=raw,value=latest,enable={{is_default_branch}}
-            type=ref,event=pr
-            type=semver,pattern={{version}}
-            type=sha,format=long
-      - name: Get committer date
+      - name: Setup and configure Docker
+        shell: 'script -q -e -c "bash {0}"'
         run: |
-          echo "committer_date=$(git log -1 --pretty=%ct)" >> "$GITHUB_ENV"
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@d70bba72b1f3fd22344832f00baa16ece964efeb # v3.3.0
+          [[ $EUID -ne 0 ]] && exec sudo /bin/sh "$0" "$@"
+          for pkg in docker.io docker-doc docker-compose docker-compose-v2 podman-docker containerd runc; \
+            do apt-get remove $pkg; \
+          done
+          apt-get update
+          apt-get install ca-certificates curl
+          install -m 0755 -d /etc/apt/keyrings
+          curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
+          chmod a+r /etc/apt/keyrings/docker.asc
+          echo \
+            "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
+            $(. /etc/os-release && echo "$VERSION_CODENAME") stable" | \
+            tee /etc/apt/sources.list.d/docker.list > /dev/null
+          apt-get update
+          cat << EOF >/etc/docker/daemon.json
+          {
+            "features": {
+            "containerd-snapshotter": true
+            },
+            "registry-mirrors": ["https://ghcr.io/tkhq"]
```

Comment:
What does this do? Why is it needed?

Reply:
We had it in mono, but I suspect it is only for pull-through cache use cases that may not be relevant here.